packages/backend-function/src/factory.ts

```diff
@@ -324,6 +332,27 @@ class FunctionFactory implements ConstructFactory<AmplifyFunction> {
     return this.props.memoryMB;
   };
 
+  private resolveEphemeralStorageSize = () => {
+    const ephemeralStorageSizeMin = 512;
+    const ephemeralStorageSizeMax = 10240;
+    const ephemeralStorageSizeDefault = 512;
+    if (this.props.ephemeralStorageSizeMB === undefined) {
+      return ephemeralStorageSizeDefault;
+    }
+    if (
+      !isWholeNumberBetweenInclusive(
+        this.props.ephemeralStorageSizeMB,
+        ephemeralStorageSizeMin,
+        ephemeralStorageSizeMax
+      )
+    ) {
+      throw new Error(
```

Comment:
Please throw an `AmplifyUserError` instead of a regular error. See line 327 for an example
https://github.com/aws-amplify/amplify-backend/pull/2283/files#diff-e354c6c4617ce06e0e16948f4521eca5f9269445265b80ed16720372d336c886R327